### PR TITLE
Allow maxprocs to debug log after start banner

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -115,15 +115,17 @@ func main() {
...
-
 	// Start things up. Block here until done.
 	if err := server.Run(s); err != nil {
 		server.PrintAndDie(err.Error())
 	}
+
+	// Adjust MAXPROCS if running under linux/cgroups quotas.
+	undo, err := maxprocs.Set(maxprocs.Logger(s.Debugf))
+	if err != nil {
+		server.PrintAndDie(fmt.Sprintf("failed to set GOMAXPROCS: %v", err))
```
Comment (ripienaar) [inline main.go:126]: is this really fatal? Seems like something we can pass over with a warning

Comment (derekcollison) [reply main.go:126]: From looking at the maxprocs this is really never expected to fail and if it does something is wrong, at least that is what I get the sense of looking at their code and all of their examples that consider it fatal.

I think we see if this ever fails in the wild, and with the server not starting we will have a higher probability of knowing about it then a warning, then adapt as needed.

**Review by ripienaar — APPROVED** [review]
LGTM

**Review by kozlovic — APPROVED** [review]
LGTM

**Pull request merged**